src/Services/QuestionnaireResponseService.php

```diff
@@ -325,113 +244,293 @@ public function parseQuestionnaireResponseForms($response): array
         return $question_results;
     }
 
-    /* WIP */
     /**
-     * @param $pid
-     * @param $foreign_id
-     * @param $name
-     * @param $q
-     * @param $response
-     * @param $form_response
+     * @param       $response
+     * @param       $pid
+     * @param null  $encounter
+     * @param null  $qr_id
+     * @param null  $qr_record_id
+     * @param null  $q
+     * @param null  $q_id
+     * @param null  $form_response
+     * @param bool  $add_report
+     * @param array $scores
      * @return array|false|int|mixed
+     * @throws Exception
      */
-    public function saveQuestionnaireResponse($pid, $foreign_id, $name, $q, $response, $form_response = null)
-    {
+    public function saveQuestionnaireResponse(
+        $response,
+        $pid,
+        $encounter = null,
+        $qr_id = null,
+        $qr_record_id = null,
+        $q = null,
+        $q_id = null,
+        $form_response = null,
+        $add_report = false,
+        $scores = []
+    ): mixed {
```

Comment:
mixed return type needs php8 (this is fine for master/7.0.1 since we will be requiring php8, however something to think about when patching this back to 7.0.0 patch which will be supporting php7.4)